R/dsl.R

```diff
@@ -784,9 +784,63 @@ list_container <- function(content_list, is_divided = FALSE) {
   list_class <- paste("ui", divided_list, "list")
   div(class = list_class,
       content_list %>% purrr::map(function(x) {
-        if (is.null(x$header) && is.null(x$descriptio))
+        if (is.null(x$header) && is.null(x$description))
           stop("content_list needs to have either header or description.")
         list_element(x$header, x$description, x$icon)
       })
   )
 }
+
+
+#' Accordion UI
+#'
+#' In accordion you may display a list of elements that can be hidden or
+#' shown with one click.
+#'
+#' @param accordion_list lsit with lists with fields: `title` and `content`
+#' @param fluid if accordin is fluid then it takes width of parent div
+#' @param active_title if active title matches `title` from \code{accordion_list}
+#' then this element is active by default
+#' @param styled if switched of then raw style (no boxes) is used
+#'
+#' @return shiyn tag list with accordion UI
+#' @export
+#'
+#' @examples
+#' if (interactive()) {
+#' library(shiny)
+#' library(shiny.semantic)
+#' accordion_content <- list(
+#'   list(title = "AA", content = h2("a a a a")),
+#'   list(title = "BB", content = p("b b b b"))
+#' )
+#' shinyApp(
+#'   ui = semanticPage(
+#'     accordion(accordion_content, fluid = F, active_title = "AA")
+#'   ),
+#'   server = function(input, output) {}
+#' )
+#' }
+accordion <- function(accordion_list, fluid = TRUE, active_title = NA, styled = TRUE) {
+  fluid <- ifelse(fluid, "fluid", "")
+  styled <- ifelse(styled, "styled", "")
+  accordion_class = glue::glue("ui {styled} {fluid} accordion")
```

Comment:
How about adding an argument to pass custom class to accordion?

Reply:
That's a good idea, just to be sure you mean additional style, eg:

```r
div(class = accordion_class, style = user_defined_style, ...)
```